axiom/nr_errors.c

```diff
@@ -14,37 +14,68 @@
 #include "util_strings.h"
 #include "util_time.h"
 
-nr_error_t* nr_error_create(int priority,
+static nr_error_t* nr_error_create_helper(int priority,
                             const char* message,
                             const char* klass,
+                            const char* error_file,
+                            int error_line,
+                            const char* error_context,
+                            int error_no,
                             const char* stacktrace_json,
                             const char* span_id,
                             nrtime_t when) {
   nr_error_t* error;
 
-  if (0 == message) {
-    return 0;
-  }
-  if (0 == klass) {
-    return 0;
-  }
-  if (0 == stacktrace_json) {
-    return 0;
-  }
-
   error = (nr_error_t*)nr_zalloc(sizeof(nr_error_t));
   error->priority = priority;
   error->when = when;
   error->message = nr_strdup(message);
   error->klass = nr_strdup(klass);
   error->stacktrace_json = nr_strdup(stacktrace_json);
+  if (NULL != error_file && NULL != error_context) {
```

Comment:
If only one of these two values is defined do we still want to create the associated field?

Reply:
After discussion, it has been decided that if one attribute is passed in as `NULL`, the rest of the attributes should still show up. I will create a commit for this.